src/runtime-rs/crates/resource/src/volume/hugepage.rs

```diff
@@ -0,0 +1,219 @@
+    io::{BufRead, BufReader},
+};
+
+use agent::Storage;
+use anyhow::{anyhow, Context, Result};
+use byte_unit::Byte;
+use kata_sys_util::fs::get_base_name;
+use kata_types::mount::KATA_EPHEMERAL_VOLUME_TYPE;
+
+use crate::share_fs::EPHEMERAL_PATH;
+
+use super::Volume;
+
+const PROC_MOUNTS_FILE: &str = "/proc/mounts";
+
+pub(crate) struct Hugepage {
+    storage: Option<Storage>,
+    mount: oci::Mount,
+}
+
+/// handle hugepage
+impl Hugepage {
+    pub(crate) fn new(
+        m: &oci::Mount,
+        options_map: Option<HashMap<Byte, u64>>,
+        fs_options: Option<Vec<String>>,
+    ) -> Result<Self> {
+        // Create mount option string
+        let option = if let Some(page_size) = get_page_size(fs_options) {
+            let page_size =
+                Byte::from_str(page_size).context("failed to create Byte object from String")?;
+            options_map
+                .context("failed to get opetions map")?
+                .get(&page_size)
+                .map(|size| format!("pagesize={},size={}", page_size.get_bytes(), size))
+        } else {
+            None
+        }
+        .context("failed to get huge page options")?;
+        let mut mount = m.clone();
+        let base_name = get_base_name(m.source.clone())?
+            .into_string()
+            .map_err(|e| anyhow!("{:?}", e))?;
+        // Set the mount source path to a path that resides inside the VM
+        mount.source = format!("{}{}{}", EPHEMERAL_PATH, "/", base_name);
+        // Set the mount type to "bind"
+        mount.r#type = "bind".to_string();
+
+        // Create a storage struct so that kata agent is able to create
+        // hugetlbfs backed volume inside the VM
+        let storage = Storage {
+            driver: KATA_EPHEMERAL_VOLUME_TYPE.to_string(),
+            source: "nodev".to_string(),
+            fs_type: "hugetlbfs".to_string(),
+            mount_point: mount.source.clone(),
+            options: vec![option],
+            ..Default::default()
+        };
+        Ok(Self {
+            storage: Some(storage),
+            mount,
+        })
+    }
+}
+
+impl Volume for Hugepage {
+    fn get_volume_mount(&self) -> Result<Vec<oci::Mount>> {
+        Ok(vec![self.mount.clone()])
+    }
+
+    fn get_storage(&self) -> Result<Vec<agent::Storage>> {
+        let s = if let Some(s) = self.storage.as_ref() {
+            vec![s.clone()]
+        } else {
+            vec![]
+        };
+        Ok(s)
+    }
+
+    fn cleanup(&self) -> Result<()> {
+        todo!()
+    }
+}
+
+pub(crate) fn is_huge_page(m: &oci::Mount) -> Result<(bool, Option<Vec<String>>)> {
+    if m.source.is_empty() {
+        return Err(anyhow!("empty mount source"));
+    }
+    let file = File::open(PROC_MOUNTS_FILE).context("failed open file")?;
+    let reader = BufReader::new(file);
+    for line in reader.lines().flatten() {
+        let items: Vec<&str> = line.split(' ').collect();
+        if m.source == items[1].to_string() && items[2] == "hugetlbfs" {
```

Comment:
`if m.source.as_ref() == items[1].to_string()` is the correct way to do this.

Reply:
fixed